### PR TITLE
Cnv 1602

### PR DESCRIPTION
**Pull request by bgaydosrh**
Adding assembly and modules for "Creating Virtual Machines" (CNV-1602) Jira story.

- I have gotten feedback from @aburdenthehand (Andrew) to delete the NOTE in the assembly (cnv-create-vms.adoc) as the Template info will be in another assembly altogether.

- @alexxa (Irina) had opened a bug (https://bugzilla.redhat.com/show_bug.cgi?id=1678782) against this content's presentation, so I anticipate she will have feedback as well.

- This assembly contains a number of reference tables. Right now they are all Level 2 headings under the last module (Using CLI to Create a VM), indicating they could apply to all three use cases (GUI, YAML, CLI). However, one who knows the content better may have a suggestion as to which reference modules best support each of the three use cases. Please let me know your recommendations as to whether they all should remain at the end (perhaps moved to Level 1) or structured differently.

**This PR is targeted for 2.0 release.

I will need a resource to merge the content, so please volunteer if you can do this for me. I will squash my commits prior.**

Thanks

Bob

### CODE REVIEW

**Comment by aburdenthehand** [comment]
@bgaydosrh - A few things jumped out at me that would be good to fix before this gets reviewed:
1) The following files/folder need to be removed:
 cnv_users_guide/modules/cnv_creating_template_wizard_web.adoc
 cnv_users_guide/modules/cnv_creating_vm.adoc
 cnv_users_guide/modules/cnv_creating_vm_wizard_web.adoc
 cnv_users_guide/modules/cnv_creating_vm_yaml_web.adoc 
+
 modules/cnv_creating_vm.adoc
Those look like 1.4 structure/content

2) You should also probably remove:
 modules/cnv-creating-template-wizard-web.adoc
 modules/cnv-template-wizard-fields-web.adoc
 modules/cnv-supported-virtio-drivers.adoc
 modules/cnv-vm-actions-web.adoc
These are files covered by other stories and added in separate PRs. 
On a similar note: 
Drop the include for creating templates from  cnv/cnv_users_guide/cnv-create-vms.adoc
It is covered in the Template story.

3) All of the [[cnv-module-name]] IDs need to be updated to the 4.1 style: [id="cnv-module-name_{context}"]
For better explanation, see: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#module-file-metadata

4) Remove all xrefs from modules. You can only xref from one assembly to another assembly, and the xref needs to go back to the top level and include the anchor ID. e.g. xref:../../cnv/cnv_users_guide/cnv-assembly-name.adoc#cnv-assembly-anchor[published text]
There are a few examples in the cnv_users_guide you can check out for reference.
For better explanation, see: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references

5) Heading levels of all modules should be single level:
= Heading name
That way it can be included in any assembly. It is in the assembly that it's heading level is determined (by the [leveloffset=X] suffix)
For better explanation, see: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

6) Update the
// Module included in the following assemblies:
//
// * cnv_users_guide/cnv_users_guide.adoc
headings of the modules to point to the `cnv_users_guide/cnv-create-vms.adoc` assembly

Feel free to hit me up if you have any questions

**Comment by kalexand-rh** [comment]
+1 to @aburdenthehand. @bgaydosrh, please address these comments and squash. After you do, I'll review.

File: modules/cnv-creating-vm.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+The `spec` object of the VirtualMachine configuration file references
+the virtual machine settings, such as the number of cores and the amount
+of memory, the disk type, and the volumes to use.
+
+Attach the virtual machine disk to the virtual machine by referencing
+the relevant PVC `claimName` as a volume.
+
+[NOTE]
+====
+https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/[ReplicaSet]
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: Can I have a bit more context?

File: modules/cnv-creating-vm.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+is not currently supported in {ProductName}.
+====
+
+See the Reference section for information about volume types and
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: What Reference section?

File: modules/cnv-creating-vm.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+sample configuration files.
+
+.Domain settings
+|===
+|Setting | Description
+
+|cores
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: Check the cases of the settings - usually they're sentence case.

File: modules/cnv-creating-vm.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+|The number of cores inside the virtual machine. Must be a value greater than or equal to 1.
+
+|memory
+| The amount of RAM allocated to the virtual machine by the node. Specify the denomination with *M* for Megabyte or *Gi* for Gigabyte.
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: s/allocated/that is allocated
~ Specify the denomination with *M* for Megabyte or *Gi* for Gigabyte.~ Specify a value in megabytes (M) or gigabytes (Gi). 
Are you sure these aren't megabits and gigabits?

File: modules/cnv-creating-vm.adoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+|disks: name
+|The Name of the volume which is referenced. Must match the name of a volume.
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: Check the capitalization here.
s/which/that

File: modules/cnv-creating-vm.adoc [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+|===
+
+.Volume settings
+|===
+|Setting | Description
+
+|name
+|The Name of the volume. Must be a DNS_LABEL and unique within the virtual machine.
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: \`DNS_LABEL\`?

Check the capitalization of "Name."

File: modules/cnv-creating-vm.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+|persistentVolumeClaim
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: Check the capitalization - it's usually PersistentVolumeClaim.

File: modules/cnv-creating-vm.adoc [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+[NOTE]
+====
+https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/[ReplicaSet]
+is not currently supported in {ProductName}.
+====
+
+See the Reference section for information about volume types and
+sample configuration files.
+
+.Domain settings
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: Move all of these references to a different module - you can't have this much reference material as the intro to a procedure module.

File: modules/cnv-creating-vm.adoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+|The Name of the volume. Must be a DNS_LABEL and unique within the virtual machine.
+
+|persistentVolumeClaim
+| The PVC to attach to the virtual machine. The `claimName` of the PVC must be in the same project as the virtual machine.
+|===
+
+See the
+https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachinespec[kubevirt
+API Reference] for a definitive list of virtual machine settings.
+
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: You need to add ".Procedure" on a separate line to introduce the step.

File: modules/cnv-creating-vm.adoc [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+To create a virtual machine with the {product-title} client:
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: s/To/* To

File: modules/cnv-creating-vm.adoc [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: add "+" to this line.

File: modules/cnv-creating-vm.adoc [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+----
+$ oc create -f <vm.yaml>
+----
+
+Virtual machines are created in a *Stopped* state. Run a virtual machine
```
Comment (kalexand-rh) [inline modules/cnv-creating-vm.adoc]: s/Stopped/stopped

Then is starting the VM a next step?

File: modules/cnv-networking-wizard-fields-web.adoc [hunk 13]
```diff
@@ -1,9 +1,11 @@
 // Module included in the following assemblies:
 //
-// * cnv/cnv_users_guide/cnv-attaching-vm-multiple-networks.adoc
+// * cnv_users_guide/cnv-create-vms.adoc
```
Comment (kalexand-rh) [inline modules/cnv-networking-wizard-fields-web.adoc]: The "cnv/" part of the path was accurate. You should add it to all of your new modules.

File: modules/cnv-storage-wizard-fields-web.adoc [hunk 14]
```diff
@@ -0,0 +1,29 @@
+// Module included in the following assemblies:
```
Comment (kalexand-rh) [inline modules/cnv-storage-wizard-fields-web.adoc]: This is cnv-storage-wizard-fields-web.adoc on https://github.com/openshift/openshift-docs/pull/15530. Why are so many of these modules duplicated?

File: modules/cnv-supported-virtio-drivers.adoc [hunk 15]
```diff
@@ -1,32 +1,53 @@
 // Module included in the following assemblies:
 //
+<<<<<<< HEAD
```
Comment (kalexand-rh) [inline modules/cnv-supported-virtio-drivers.adoc]: You need to fix these merge conflict remnants. I'm not reviewing this content because I don't know what you're keeping.

File: modules/cnv-networking-wizard-fields-web.adoc [hunk 16]
```diff
@@ -1,9 +1,11 @@
 // Module included in the following assemblies:
 //
-// * cnv/cnv_users_guide/cnv-attaching-vm-multiple-networks.adoc
```
Comment (ousleyp) [inline modules/cnv-networking-wizard-fields-web.adoc]: Why did you remove this? The module is included in that assembly, too :)

File: modules/cnv-vm-storage-volume-types.adoc [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+[id="cnv-vm-storage-volume-types_{context}"]
+= Types of storage volumes for virtual machines
+
+[horizontal]
+*ephemeral*::
+A local copy-on-write (COW) image that uses a network volume as a
+read-only backing store. The backing volume
+must be a *PersistentVolumeClaim*. The ephemeral image is created when
+the virtual machine starts, and stores all writes locally. The ephemeral
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: no comma

File: modules/cnv-vm-storage-volume-types.adoc [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+image is discarded when the virtual machine is stopped, restarted, or
+deleted. The backing volume (PVC) is not mutated in any way.
+
+*persistentVolumeClaim*::
+Attaches an available PV to a virtual machine. This allows for the
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: Add a noun after "This."

File: modules/cnv-vm-storage-volume-types.adoc [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+virtual machine data to persist between sessions.
++
+Importing an existing virtual machine disk into a PVC using
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: s/using/by using

File: modules/cnv-vm-storage-volume-types.adoc [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+CDI and attaching the PVC to a virtual machine instance is the
+recommended method for importing existing virtual machines into
+{product-title}. There are some requirements for the disk to be used within a
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: Where are these requirements?

File: modules/cnv-vm-storage-volume-types.adoc [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+PVC.
+
+*dataVolume*::
+DataVolumes build on the *persistentVolumeClaim* disk type by managing the process
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: Why persistentVolumeClaim?

File: modules/cnv-vm-storage-volume-types.adoc [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+of preparing the virtual machine disk via an import, clone, or upload operation.
+VMs using this volume type are guaranteed not to start until the volume is ready.
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: s/using/that use

File: modules/cnv-vm-storage-volume-types.adoc [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+
+*cloudInitNoCloud*::
+Attaches a disk containing the referenced cloud-init NoCloud data
```
Comment (kalexand-rh) [inline modules/cnv-vm-storage-volume-types.adoc]: s/containing/that contains

**Review by kalexand-rh — COMMENTED** [review]
You have some duplicate modules with other in-flight PRs and some unresolved merge errors. I have some other picks and suggestions before this is ready to merge.

File: modules/cnv-supported-virtio-drivers.adoc (hunk 15 above)
Comment (vikram-redhat) [reply modules/cnv-supported-virtio-drivers.adoc]: @bgaydosrh if this module is not included in any assembly, you should remove this.

Comment (aburdenthehand) [reply modules/cnv-supported-virtio-drivers.adoc]: @bgaydosrh - There's no reason for this module to be included in this PR.

File: modules/cnv-creating-vm-wizard-web.adoc [hunk 24]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
```
[the remark this answers is not in the document]
Comment (kalexand-rh) [reply modules/cnv-creating-vm-wizard-web.adoc]: Actually, it's not. Andrew pointed out that one's the VM and one's the VM template, and that they're just so similar that they should be combined and conditionalized on the same PR.

File: modules/cnv-networking-wizard-fields-web.adoc (hunk 16 above)
Comment (bgaydosrh) [reply modules/cnv-networking-wizard-fields-web.adoc]: Hi Pan, my mistake. I can re-include it.

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 25]
```diff
@@ -0,0 +1,20 @@
+[id="create-vms"]
+= Creating Virtual Machines
+include::modules/cnv_document_attributes.adoc[]
+:context: creating-virtual-machines
+toc::[]
+
+Create a virtual machine by:
+
+* Running the GUI wizard
+* Pasting in a pre-configured YAML file
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: Should I clarify the GUI is in the Web Console or steer clear of the term GUI altogether? Thanks.

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 26]
```diff
@@ -0,0 +1,20 @@
+[id="create-vms"]
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: Resolved. Changed to [id="cnv-create-vms"]

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+= Creating Virtual Machines
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: Resolved. Changed to "Creating virtual machines"

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 28]
```diff
@@ -0,0 +1,20 @@
...
+include::modules/cnv_document_attributes.adoc[]
+:context: creating-virtual-machines
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: Resolved. Changed to :context: cnv-create-vms.

File: cnv/cnv_users_guide/cnv-create-vms.adoc (hunk 25 above)
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: I changed this to "web console wizard".

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+toc::[]
+
+Create a virtual machine by:
+
+* Running the GUI wizard
+* Pasting in a pre-configured YAML file
+* Using the CLI
+
+include::modules/cnv-creating-vm-wizard-web.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: I changed my intro to read:

**Create a virtual machine by following one of these three procedures:

* Running the virtual machine wizard
* Pasting a pre-configured YAML file with the virtual machine wizard
* Using the CLI**

I changed the procedure module names to match these bullets, appending "to create a virtual machine" to the headings so the modules can be reused with full context.

I added "Procedure" after the heads of the Procedure modules.

[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: I also added sentences to indicate which reference modules should be used in each case, such as:

"Refer to the Virtual machine wizard fields section when running the web console wizard."

[the remark this answers is not in the document]
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: I am creating a bug to address technical questions with SMEs that can be resolved at a later time so these issues do not hold up the release. I will ask for verification on the usage of these ref modules with each procedure.

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 30]
```diff
@@ -0,0 +1,33 @@
+[id="cnv-create-vms"]
+= Creating virtual machines
+include::modules/cnv_document_attributes.adoc[]
+:context: cnv-create-vms
+toc::[]
```
Comment (kalexand-rh) [inline cnv/cnv_users_guide/cnv-create-vms.adoc]: Your TOC isn't rendering for me.

File: cnv/cnv_users_guide/cnv-create-vms.adoc [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+
+Refer to the Virtual machine wizard fields section when running the web console wizard.
+
+include::modules/cnv-create-vms-ref-table-wizard-fields-web.adoc[leveloffset=+1]
+
+include::modules/cnv-creating-vm-yaml-web.adoc[leveloffset=+1]
+
+include::modules/cnv-creating-vm.adoc[leveloffset=+1]
+
+Refer to the following tables when creating a virtual machine with the CLI:+
```
Comment (kalexand-rh) [inline cnv/cnv_users_guide/cnv-create-vms.adoc]: Remove the +

File: modules/cnv-networking-wizard-fields-web.adoc (hunk 16 above)
Comment (bgaydosrh) [reply modules/cnv-networking-wizard-fields-web.adoc]: It should be back in now.

File: modules/cnv-cloud-init-fields-web.adoc [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+// * cnv_users_guide/cnv-create-vms.adoc
+
+[id="cnv-cloud-init-fields-web_{context}"]
+= Cloud-init fields
+
+|===
+|Name | Description
+
+|Hostname
+|Sets a specific hostname for the virtual machine.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-cloud-init-fields-web.adoc]: Resolved.

File: modules/cnv-cloud-init-fields-web.adoc [hunk 33]
```diff
@@ -0,0 +1,19 @@
...
+
+|Authenticated SSH Keys
+|The user's public key. This will be copied to *_~/.ssh/authorized_keys_* on the virtual machine.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-cloud-init-fields-web.adoc]: Changed to:

The user's public key that is copied to *_~/.ssh/authorized_keys_* on the virtual machine.

(unless you prefer which with the comma)

File: modules/cnv-cloud-init-fields-web.adoc [hunk 34]
```diff
@@ -0,0 +1,19 @@
...
+
+|Use custom script
+|Replaces other options with a textbox into which you can paste a custom cloud-init script.
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-cloud-init-fields-web.adoc]: Changed to:

Replaces other options with a field into which you paste a custom cloud-init script.

File: modules/cnv-create-vms-ref-table-wizard-fields-web.adoc [hunk 35]
```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-create-vms-ref-table-wizard-fields-web.adoc]: Cool. Updating assembly to usecnv-vm-wizard-fields-web.adoc.

[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-create-vms-ref-table-wizard-fields-web.adoc]: Cool. Updating assembly to use cnv-vm-wizard-fields-web.adoc.

File: modules/cnv-creating-vm-wizard-web.adoc (hunk 24 above)
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-creating-vm-wizard-web.adoc]: Yes @aburdenthehand is going to conditionalize that. 

@aburdenthehand, If I have action items on this, just let me know.

File: modules/cnv-creating-vm-yaml-web.adoc [hunk 36]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-creating-vm-yaml-web.adoc]: I think this is the same situation as the above comment for cnv-creating-vm-wizard-web. @aburdenthehand can you confirm?

File: modules/cnv-creating-vm.adoc (hunk 1 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Changed Note to read:
"A  ReplicaSet is often used to guarantee the availability of a specified number of identical Pods.
ReplicaSet is not currently supported in {ProductName}."

File: modules/cnv-creating-vm.adoc (hunk 2 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Changed to:

"See the virtual machine storage volume types section for information about volume types."

I will query QE on the availability and location of sample config files and if we want to include them.

File: modules/cnv-creating-vm.adoc (hunk 3 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: I changed the casing for Cores, Memory, Disks: name, Name.

But the last setting, persistentVolumeClaim, appears to be a programmatic string var, and probably case dependent. Which leads me to believe they all may be.

@kalexand-rh what do you think?

File: modules/cnv-creating-vm.adoc (hunk 2 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: On second thought, it seems a better option to use an include to bring this table into this CLI module at the needed point. Is this allowed?

When I try using include::modules/cnv-vm-storage-volume-types.adoc[leveloffset=+1],
this is rendering with the '[leveloffset]' attribute appearing before the table.

@kalexand-rh what do you recommend? Thanks.

File: modules/cnv-creating-vm.adoc (hunk 4 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Changed to:

"Memory
|The amount of RAM that is allocated to the virtual machine by the node. Specify a value in *M* for Megabyte or *Gi* for Gigabyte."

I will query QE on bit vs. byte.

File: modules/cnv-creating-vm.adoc (hunk 5 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Lower-cased name; using 'that'.

"The name of the volume that is referenced. Must match the name of a volume."

I'm not sure I would keep the second sentence. It seems redundant.

File: modules/cnv-creating-vm.adoc (hunk 6 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Changed to:

The name of the volume, which must be a DNS label and unique within the virtual machine.

(**DNS label** is proper for the noun; I see no reason to use the var name as it is plainly understandable. This seems to be the convention in online mentions of the term.)

File: modules/cnv-creating-vm.adoc (hunk 7 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Ah, OK. Then if that is so, my prior question on the casing is moot. I will flag it for QE to check.

File: modules/cnv-creating-vm-wizard-web.adoc (hunk 24 above)
[the remark this answers is not in the document]
Comment (aburdenthehand) [reply modules/cnv-creating-vm-wizard-web.adoc]: @bgaydosrh Best to merge this first. I'll create a new PR to combine the files once it's all in master.

File: modules/cnv-creating-vm.adoc (hunk 3 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Never mind. Resolved by one of your subsequent comments :-)

File: modules/cnv-cloud-init-fields-web.adoc [hunk 37]
```diff
@@ -0,0 +1,19 @@
...
+//
+// * cnv_users_guide/cnv-create-vms.adoc
+
+[id="cnv-cloud-init-fields-web_{context}"]
+= Cloud-init fields
+
+|===
+|Name | Description
+
+|Hostname
```
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-cloud-init-fields-web.adoc]: Will query QE regarding this.

File: modules/cnv-creating-vm.adoc (hunk 8 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Good point. I moved the actual steps (and numbered them) to the top:

.Procedure

The `spec` object of the VirtualMachine configuration file references
the virtual machine settings, such as the number of cores and the amount
of memory, the disk type, and the volumes to use.

. Attach the virtual machine disk to the virtual machine by referencing
the relevant PVC `claimName` as a volume.

. To create a virtual machine with the {product-title} client, run this command:

----
$ oc create -f <vm.yaml>
----

. Since virtual machines are created in a *Stopped* state, run a virtual machine
instance by starting it.

File: modules/cnv-creating-vm.adoc (hunk 9 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: I combined this reference intro with the line that introduces the storage volume types ref module and reworded it a bit to include the the domain and volume settings which follow in-line. The Procedure is at the top as suggested.

The ref intro reads (again not sure I can include a module from a module)

Virtual machine storage volume types are listed here, as well as domain and volume settings. See the
https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachinespec[kubevirt
API Reference] for a definitive list of virtual machine settings.
include::modules/cnv-vm-storage-volume-types.adoc[leveloffset=+1]

File: modules/cnv-creating-vm.adoc (hunk 10 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Done.

File: modules/cnv-creating-vm.adoc (hunk 11 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Done.

File: modules/cnv-creating-vm.adoc (hunk 12 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: I changed this to the third step in the Procedure.

The whole procedure is:

. Attach the virtual machine disk to the virtual machine by referencing
the relevant PVC `claimName` as a volume.

. To create a virtual machine with the {product-title} client, run this command:
+
----
$ oc create -f <vm.yaml>
----

. Since virtual machines are created in a *Stopped* state, run a virtual machine
instance by starting it.

File: modules/cnv-networking-wizard-fields-web.adoc (hunk 13 above)
Comment (bgaydosrh) [reply modules/cnv-networking-wizard-fields-web.adoc]: Done.

File: modules/cnv-creating-vm.adoc (hunk 9 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Ah, I see now this is not possible. What I will do is move the reference intro into the assembly and then call storage type table from the assembly. I think I will make modules of the domain and volume settings, too.

File: modules/cnv-creating-vm.adoc (hunk 2 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: See my comment below. Realized you can't call from a module. Doh!

File: modules/cnv-creating-vm-wizard-web.adoc (hunk 24 above)
[the remark this answers is not in the document]
Comment (bgaydosrh) [reply modules/cnv-creating-vm-wizard-web.adoc]: cool beans. thx @aburdenthehand

File: modules/cnv-storage-wizard-fields-web.adoc (hunk 14 above)
Comment (bgaydosrh) [reply modules/cnv-storage-wizard-fields-web.adoc]: Not sure what this comment means. Is there another version of this module used in 15530?

File: cnv/cnv_users_guide/cnv-create-vms.adoc (hunk 30 above)
Comment (bgaydosrh) [reply cnv/cnv_users_guide/cnv-create-vms.adoc]: Let's follow-up on this one after the topic map file is updated early this week. I would like to get the outstanding PRs merged early Monday. There will definitely be more changes following QE review but I'd like to get the largest amount of changes merged sooner than later.

File: modules/cnv-supported-virtio-drivers.adoc (hunk 15 above)
Comment (bgaydosrh) [reply modules/cnv-supported-virtio-drivers.adoc]: Hi Vikram -

At the moment I am unsure how to explicitly remove a file with the old git version I am using (1.8). In this version you must explicitly tell git to forget about the file with "git add <filename>" after removing it from your local branch. Previously I was using the -D argument but believe that now to be bad practice (nothing else was working at the time). Should I try with -d argument, so "git add modules/cnv-supported-virtio-drivers.adoc -d"?

File: modules/cnv-vm-storage-volume-types.adoc (hunk 17 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Done. Nice catch.

File: modules/cnv-vm-storage-volume-types.adoc (hunk 18 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Changed to:

"Attaches an available PV to a virtual machine. Attaching a PV allows for the
virtual machine data to persist between sessions."

File: modules/cnv-vm-storage-volume-types.adoc (hunk 19 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Changed to:

Importing an existing virtual machine disk into a PVC by using

File: modules/cnv-vm-storage-volume-types.adoc (hunk 20 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Flagged for QE review.

File: modules/cnv-vm-storage-volume-types.adoc (hunk 21 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: I believe there is dependence between datavolume and persistentvolumeclaim but I will clarify with QE and make this plainer , if so.

File: modules/cnv-vm-storage-volume-types.adoc (hunk 22 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Changed to:

"VMs that use this volume type are guaranteed not to start until the volume is ready."

File: modules/cnv-vm-storage-volume-types.adoc (hunk 23 above)
Comment (bgaydosrh) [reply modules/cnv-vm-storage-volume-types.adoc]: Changed to:

"Attaches a disk that contains the referenced cloud-init NoCloud data
source, providing user data and metadata to the virtual machine.
A cloud-init installation is required inside the virtual machine
disk."

File: modules/cnv-creating-vm.adoc (hunk 9 above)
Comment (bgaydosrh) [reply modules/cnv-creating-vm.adoc]: Upon further thought, I am not breaking out the domain and volume settings from the CLI topic as I don't know at this point what other assemblies call in the CLI task module.

So, I have finalized on this solution, with the procedure at the top:

[id="cnv-create-vms"]
= Creating virtual machines
include::modules/cnv_document_attributes.adoc[]
:context: cnv-create-vms
toc::[]

Use one of these procedures to create a virtual machine:

* Running the virtual machine wizard
* Pasting a pre-configured YAML file with the virtual machine wizard
* Using the CLI

include::modules/cnv-creating-vm-wizard-web.adoc[leveloffset=+1]

Refer to the Virtual machine wizard fields section when running the web console wizard.

include::modules/cnv-vm-wizard-fields-web.adoc[leveloffset=+1]

include::modules/cnv-creating-vm-yaml-web.adoc[leveloffset=+1]

include::modules/cnv-creating-vm.adoc[leveloffset=+1]

Virtual machine storage volume types are listed here, as well as domain and volume settings. See the
https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachinespec[kubevirt
API Reference] for a definitive list of virtual machine settings.

Refer to the following tables when creating a virtual machine with the CLI:

* Storage volume types
* Cloud-init fields
* Networking fields
* Storage fields

include::modules/cnv-vm-storage-volume-types.adoc[leveloffset=+1]
include::modules/cnv-cloud-init-fields-web.adoc[leveloffset=+1]
include::modules/cnv-networking-wizard-fields-web.adoc[leveloffset=+1]
include::modules/cnv-storage-wizard-fields-web.adoc[leveloffset=+1]

**Comment by bgaydosrh** [comment]
Squashed and re-pushed to 1 commit

**Comment by vikram-redhat** [comment]
Merging.
